### PR TITLE
Name Translation Schema Changes

### PR DESCRIPTION
**Pull request by lekshmimallika-aot**
*Issue #:* /bcgov/entity#2463

*Description of changes:*
Added name translations to the incorporation schema
Added tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-schemas license (Apache 2.0).

### CODE REVIEW

File: src/registry_schemas/example_data/schema_data.py [hunk 1]
```diff
@@ -762,6 +762,8 @@
     'nameRequest': {
         'legalType': 'BC'
     },
+    'hasNameTranslations': True,
```
Comment (kialj876) [inline src/registry_schemas/example_data/schema_data.py:765]: Everything looks good to me so I will approve, but is there a reason to have this boolean? Wouldn't it be the same logic to have if json.get('nameTranslations') in any of our FE or BE code as if json['hasNameTranslations']?

File: src/registry_schemas/schemas/incorporation.json [hunk 2]
```diff
@@ -43,7 +44,34 @@
         },
         "incorporationAgreement": {
             "$ref": "https://bcrs.gov.bc.ca/.well_known/schemas/agreementType"
+        },
+        "hasNameTranslations": {
```
Comment (severinbeauvais) [inline src/registry_schemas/schemas/incorporation.json:48]: The Create UI declares IncorporationFilingIF, which should be updated to include this new boolean and array.

Comment (lekshmimallika-aot) [reply src/registry_schemas/schemas/incorporation.json:48]: @cameron-freshworks is working on that

File: src/registry_schemas/example_data/schema_data.py (hunk 1 above)
Comment (lekshmimallika-aot) [reply src/registry_schemas/example_data/schema_data.py]: Removed that property

**Pull request merged**